code/modules/robotics/bot/secbot.dm

```diff
@@ -1198,18 +1494,121 @@ Report Arrests: <A href='?src=\ref[src];operation=report'>[report_arrests ? "On"
 				user.u_equip(W)
 				qdel(W)
 
-	else if (istype(W, /obj/item/cable_coil) && src.build_step >= 4)
+	else if (istype(W, /obj/item/cable_coil) && src.build_step == 4)
 		var/obj/item/cable_coil/C = W
 		if (!C.use(5))
 			boutput(user, "You need a longer length of cable! A length of five should be enough.")
 		else
 			src.build_step++
-			boutput(user, "You add the wires to the rod, completing the Securitron! Beep boop.")
-			var/obj/machinery/bot/secbot/S = new /obj/machinery/bot/secbot(get_turf(src))
-			S.beacon_freq = src.beacon_freq
-			S.hat = src.hat
-			S.name = src.created_name
-			qdel(src)
+			boutput(user, "You add the wires to the rod, completing the Securitron's stun baton.")
+			src.name = "helmet/signaler/prox sensor/robot arm/stun rod assembly"
+			src.overlays += image('icons/obj/bots/aibots.dmi', "hs_stun")
+			C.use(5)
+
+	else if (istype(W, /obj/item/parts/robot_parts/arm/) && src.build_step == 5)
+		src.build_step++
+		boutput(user, "You add another robot arm to [src]!")
+		src.name = "helmet/signaler/prox sensor/robot arms/stun rod assembly"
+		src.overlays += image('icons/obj/bots/aibots.dmi', "hs_arms")
+		user.u_equip(W)
+		qdel(W)
+
+	else if (istype(W, /obj/item/gun/energy/taser_gun) && src.build_step >= 5)
+		src.build_step++
+		boutput(user, "You place the taser gun into the Securitron's robotic hand, activating it! Beep boop.")
+		var/obj/machinery/bot/secbot/S = new /obj/machinery/bot/secbot(get_turf(src))
+		S.beacon_freq = src.beacon_freq
+		S.hat = src.hat
+		S.name = src.created_name
+		S.botgun = "taser"
+		S.gunhat = "taser"
+		qdel(W)
+		qdel(src)
```

Comment:
Lots of duplicate code here and below.

What you could do is store the reference to the gun you're adding (without actually deleting it), then scrape whatever information you need from that, such as which projectile type to use, etc. This also has the benefit of removing this feature as a source of weapon duplication (see my earlier comment).

If you're concerned about people then being able to add any type of gun to the assembly, you can just make a whitelist for that.

Reply:
Done and done, and thanks again for walking me through all this!